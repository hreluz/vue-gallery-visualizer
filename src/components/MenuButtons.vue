<template>
  <div class="row my-3">
    <div class="col">
      <button type="button" disabled class="btn btn-outline-primary previous-button" @click="previous">Previous</button>
    </div>
    <div class="col">
      <button type="button" class="btn btn-outline-warning random-button" @click="random">Random</button>
    </div>
    <div class="col">
      <button type="button" class="btn btn-outline-primary next-button" @click="next">Next</button>
    </div>
  </div>
</template>
<script lang="ts">
import { defineComponent } from 'vue';
export default defineComponent({
  name: 'MenuButtons',
});
</script>