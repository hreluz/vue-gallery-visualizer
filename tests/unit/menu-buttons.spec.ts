import MenuButtons from "@/components/MenuButtons.vue";
import { shallowMount, VueWrapper } from '@vue/test-utils'

describe('MenuButtons.vue', () => {

  let wrapper:VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(MenuButtons);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should show the three buttons', () => {
    wrapper.find('.previous-button').exists();
    wrapper.find('.random-button').exists();
    wrapper.find('.next-button').exists();
  });

  it('should have the previous button disabled', () => {
    const previousButton = wrapper.find('.previous-button');
    const randomButton = wrapper.find('.random-button');

    expect(previousButton.attributes().disabled).toBeDefined();
    expect(randomButton.attributes().disabled).toBeUndefined();
  });
})
