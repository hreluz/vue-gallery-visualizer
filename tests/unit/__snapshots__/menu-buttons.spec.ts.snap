// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButtons.vue should match the snapshot 1`] = `
<div class="row my-3">
  <div class="col"><button type="button" disabled="" class="btn btn-outline-primary previous-button">Previous</button></div>
  <div class="col"><button type="button" class="btn btn-outline-warning random-button">Random</button></div>
  <div class="col"><button type="button" class="btn btn-outline-primary next-button">Next</button></div>
</div>
`;
